app/app.py

```diff
@@ -0,0 +1,246 @@
+import os
+import json
+import hashlib
+
+import jsonschema
+import requests
+from flask import (
+    Flask, request, jsonify, send_from_directory, render_template, abort,
+    send_file)
+from flask.ext.cache import Cache
+from flask.views import MethodView
+
+
+APP_ROOT = os.path.dirname(os.path.abspath(__file__))
+DEBUG = os.environ.get('DEBUG', False) in ('true', '1', 'y', 'yes')
+
+SCHEMA_URL = 'https://raw.githubusercontent.com/mozilla/contribute.json/master/schema.json'
+
+KNOWN_URLS_URL = 'https://raw.githubusercontent.com/mozilla/contribute.json/master/knownurls.txt'
+
+SAMPLE = """
+{
+  "name": "contribute.json",
+  "description": "Standard to describe open source projects",
+  "repository": {
+    "url": "https://github.com/mozilla/contribute.json",
+    "license": "MPL2"
+  },
+  "keywords": [
+    "JSON",
+    "Python",
+    "Flask"
+  ]
+}
+""".strip()
+
+app = Flask(__name__)
+
+CACHE_CONFIG = {'CACHE_TYPE': os.getenv('CACHE_TYPE', 'simple')}
+CACHE_SERVERS = os.getenv('CACHE_MEMCACHED_SERVERS', None)
+if CACHE_SERVERS:
+    CACHE_CONFIG['CACHE_MEMCACHED_SERVERS'] = [CACHE_SERVERS]
+
+cache = Cache(app, config=CACHE_CONFIG)
+
+
+def cache_set(key, value, *args, **options):
+    if isinstance(value, (dict, list, bool)):
+        value = json.dumps(value)
+    cache.set(key, value, *args, **options)
+
+
+def cache_get(key, default=None):
+    value = cache.get(key)
+    if value is None:
+        value = default
+    if value is not None and not isinstance(value, (dict, list, bool)):
+        value = json.loads(value)
+    return value
+
+
+@app.route('/contribute.json')
+def this_contribute_json():
+    return send_file('root_files/contribute.json')
+
+
+@app.route('/')
+def index_html():
+    return catch_all('index.html')
+
+
+@app.route('/static/<path:path>')
+def serve_static(path):
+    # should only be used when run from __main__.py.
```

Comment:
no. there's still a `__main__.py` file, it just imports from here. It still uses this because when run via `__main__.py` it doesn't use Whitenoise.